documentation/docs/30-advanced/40-service-workers.md

```diff
@@ -4,10 +4,73 @@ title: Service workers
 
 Service workers act as proxy servers that handle network requests inside your app. This makes it possible to make your app work offline, but even if you don't need offline support (or can't realistically implement it because of the type of app you're building), it's often worth using service workers to speed up navigation by precaching your built JS and CSS.
 
-In SvelteKit, if you have a `src/service-worker.js` file (or `src/service-worker.ts`, or `src/service-worker/index.js`, etc) it will be built with Vite and automatically registered. You can disable automatic registration if you need to register the service worker with your own logic (e.g. prompt user for update, configure periodic updates, use `workbox`, etc).
+In SvelteKit, if you have a `src/service-worker.js` file (or `src/service-worker.ts`, or `src/service-worker/index.js`, etc) it will be built with Vite and automatically registered. You can change the [location of your service worker](/docs/configuration#files) if you need to.
 
-> You can change the [location of your service worker](/docs/configuration#files) and [disable automatic registration](/docs/configuration#serviceworker) in your project configuration.
+You can [disable automatic registration](/docs/configuration#serviceworker) if you need to register the service worker with your own logic (e.g. prompt user for update, configure periodic updates, use `workbox`, etc). The default registration looks something like this:
```

Comment:
```suggestion
You can [disable automatic registration](/docs/configuration#serviceworker) if you need to register the service worker with your own logic (e.g. prompt user for update, configure periodic updates, [use `workbox`](https://vite-pwa-org.netlify.app/frameworks/sveltekit.html), etc). The default registration looks something like this:
```

Reply:
I don't think just adding a link here is enough. From what I understand, that plugin works completely different from SvelteKit and has zero overlap (you can't use them together). So I think it would be better to have a separate paragraph on that.

Reply:
You can use it with SvelteKit, but I don't think you'd use it with SvelteKit's service worker. This was in a sentence about disabling SvelteKit's service worker, so I was assuming that that would be disabled first before adding the plugin, but maybe it could be made clearer with something more explicit like:

```suggestion
You can [disable automatic registration](/docs/configuration#serviceworker) if you need to register the service worker with your own logic. Examples of this would include prompting the user for update, configuring periodic updates, using `workbox`, etc. You may find the [Vite PWA plugin](https://vite-pwa-org.netlify.app/frameworks/sveltekit.html) helpful in these cases as it provides these functionalities and has built-in SvelteKit support. The default registration looks something like this:
```

Reply:
I expressed myself poorly. What I mean is that you probably can't use the `$service-worker` imports with that plugin (although I haven't tested that, but from their documentation it sounds like it isn't possible), so you can either use that plugin or use SvelteKit's service worker things, but not both.

Reply:
Oh yeah. I don't think they support it. Possibly they could add it at the cost of breaking dev mode support, but at least for now they don't have any support for it

Reply:
Added a paragraph highlighting alternatives and moves the Vite PWA plugin mention there.